pkg/skaffold/build/tag/git_commit.go

```diff
@@ -89,9 +90,31 @@ func (c *GitCommit) GenerateFullyQualifiedImageName(workingDir string, imageName
 		return fmt.Sprintf("%s:%s-dirty", imageName, ref), nil
 	}
 
+	ref = gitTagToDockerTag(ref)
+
 	return fmt.Sprintf("%s:%s", imageName, ref), nil
 }
 
+func gitTagToDockerTag(text string) string {
```

Comment:
Not: I don't think there is anything got specific about this - and it doesn't say what it does: `escapeAndTruncateAsTag`?

Reply:
Or `sanitizeTag` could be another option